src/psyclone/tests/psyir/transformations/intrinsics/matmul2code_trans_test.py

```diff
@@ -210,22 +216,22 @@ def _check_ulbound(lower_bound, upper_bound, step, index):
         respectively.
 
         '''
-        assert isinstance(lower_bound, BinaryOperation)
-        assert lower_bound.operator == BinaryOperation.Operator.LBOUND
+        assert isinstance(lower_bound, IntrinsicCall)
+        assert lower_bound.intrinsic == IntrinsicCall.Intrinsic.LBOUND
         assert isinstance(lower_bound.children[0], Reference)
         assert lower_bound.children[0].symbol is array_symbol
         assert isinstance(lower_bound.children[1], Literal)
         assert (lower_bound.children[1].datatype.intrinsic ==
                 ScalarType.Intrinsic.INTEGER)
-        assert lower_bound.children[1].value == str(index)
-        assert isinstance(upper_bound, BinaryOperation)
-        assert upper_bound.operator == BinaryOperation.Operator.UBOUND
+        assert lower_bound.children[1].value == str(index+1)
```

Comment:
Could you explain this change - was it a bug (and if so, how did it work before)?

Reply:
Yes, it was a bug. The test worked before because the same bug was on the tested code, but note the new
src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py line 109

Without this the LFRic with all transformations does not produce the same output that the LFRic passthrough (I am not sure why this didn't happen before, but now more LFRic kernels have no codeblocks, can be inlinded, and can have the matmuls transformed, so it may be new code)